src/components/delete/sign_as/mod.rs

```diff
@@ -0,0 +1,205 @@
+            move |network_config| {
+                let near_social_account_id = crate::consts::NEAR_SOCIAL_ACCOUNT_ID.get(network_config.network_name.as_str())
+                    .wrap_err_with(|| format!("The <{}> network does not have a near-social contract.", network_config.network_name))?;
+
+                if let Some(remote_widgets) = crate::common::get_remote_widgets(&account_id, network_config, near_social_account_id)? {
+                    let widgets_to_remove = if widgets.is_empty() {
+                        remote_widgets.keys().cloned().collect()
+                    } else {
+                        widgets.clone()
+                        .into_iter()
+                        .filter(|widget| remote_widgets.get(widget).is_some())
+                        .collect::<Vec<_>>()
+                    };
+                    if widgets_to_remove.len() > 12 {
+                        return color_eyre::eyre::Result::Err(color_eyre::eyre::eyre!("You have specified more than 12 widgets at once."));
+                    }
+                    if widgets_to_remove.is_empty() {
+                        println!("No widgets to remove. Goodbye.");
+                        return Ok(near_cli_rs::commands::PrepopulatedTransaction {
+                            signer_id: signer_id.clone(),
+                            receiver_id: near_social_account_id.clone(),
+                            actions: vec![],
+                        });
+                    }
+                    let mut actions: Vec<near_primitives::transaction::Action> = widgets_to_remove.iter()
+                        .map(|widget|
+                            near_primitives::transaction::Action::FunctionCall(
+                                near_primitives::transaction::FunctionCallAction {
+                                    method_name: "set".to_string(),
+                                    args: serde_json::json!({
+                                        "data": serde_json::json!({
+                                            account_id.to_string(): json!({
+                                                "widget": json!({
+                                                    widget.clone(): json!({
+                                                    "metadata": json!({
+                                                        "description": Null,
+                                                        "image": json!({
+                                                            "url": Null,
+                                                        }),
+                                                        "name": Null,
+                                                        "tags": json!({
+                                                            "app": Null,
+                                                            "tag": Null,
+                                                        })
```

Comment:
This is not complete, there could be other tags. Fetch all the components into `serde_json::Value` (don't parse them into `socialdb_types`), and update the leaf values to be `null`, so if the component object looks like:

```json
  "Ideas": {
    "": "/* component code */",
    "metadata": {
      "description": "A place for ideas, where they are proposed, discussed, validated, sponsored, and implemented.",
      "image": {
        "ipfs_cid": "bafkreifynesvgrnghleag5c7vgshkp7h4uegzwdhjyjqqans4t3hdpuhii"
      },
      "name": "Developer Governance Gigs",
      "tags": {
        "developer-governance": ""
      }
    }
  },
```

The Function call needs to `null` all the leaf nodes (`""`, `"metadata"."description"`, `"metadata".image"."ipfs_cid"`, ...):

```json
  "Ideas": {
    "": null,
    "metadata": {
      "description": null,
      "image": {
        "ipfs_cid": null
      },
      "name": null,
      "tags": {
        "developer-governance": null
      }
    }
  },
```

We could have deleted empty objects (`"metadata"`, `"image"`, `"tags"`), but I checked [SocialDB code](https://github.com/NearSocial/social-db/blob/aa7fafaac92a7dd267993d6c210246420a561370/contract/src/api.rs#L311) and this will only add `""` key and mark it deleted, so we only grow the storage instead of shrinking it down.

Thus, we only need a single Function Call, so we can probably remove up to 25 components at a time now, given the gas restriction.

Reply:
Done